src/main/java/mate/academy/bookstore/repository/book/BookSpecificationBuilderImpl.java

```diff
@@ -0,0 +1,40 @@
+package mate.academy.bookstore.repository.book;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.BookSearchParametersDto;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.repository.SpecificationProviderManager;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilderImpl implements BookSpecificationBuilder {
+    private static final String TITLE_KEY = "title";
+    private static final String AUTHOR_KEY = "author";
+    private static final String PRICE_KEY = "price";
+    private final SpecificationProviderManager<Book> specificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParametersDto bookSearchParametersDto) {
+        Specification<Book> spec = Specification.where(null);
+        if (bookSearchParametersDto.getTitles() != null
```

Comment:
code duplication, it is possible to create method for null and length check and reuse it or use already existing one like ArrayUtils.isEmpty